src/main/java/com/esri/core/geometry/OperatorImportFromWkbLocal.java

```diff
@@ -195,186 +199,69 @@ public OGCStructure executeOGC(int importFlags, ByteBuffer wkbBuffer,
-					wkbHelper);
-
-		case WkbGeometryType.wkbLineStringZM:
-			if (type.value() != Geometry.GeometryType.Polyline
-					&& type.value() != Geometry.GeometryType.Unknown)
-				throw new GeometryException("invalid shape type");
-			return importFromWkbPolyline(false, importFlags, true, true,
-					wkbHelper);
-
-		case WkbGeometryType.wkbMultiLineString:
-			if (type.value() != Geometry.GeometryType.Polyline
-					&& type.value() != Geometry.GeometryType.Unknown)
-				throw new GeometryException("invalid shape type");
-			return importFromWkbPolyline(true, importFlags, false, false,
-					wkbHelper);
-
-		case WkbGeometryType.wkbMultiLineStringM:
-			if (type.value() != Geometry.GeometryType.Polyline
-					&& type.value() != Geometry.GeometryType.Unknown)
-				throw new GeometryException("invalid shape type");
-			return importFromWkbPolyline(true, importFlags, false, true,
-					wkbHelper);
-
-		case WkbGeometryType.wkbMultiLineStringZ:
-			if (type.value() != Geometry.GeometryType.Polyline
-					&& type.value() != Geometry.GeometryType.Unknown)
-				throw new GeometryException("invalid shape type");
-			return importFromWkbPolyline(true, importFlags, true, false,
-					wkbHelper);
-
-		case WkbGeometryType.wkbMultiLineStringZM:
-			if (type.value() != Geometry.GeometryType.Polyline
-					&& type.value() != Geometry.GeometryType.Unknown)
-				throw new GeometryException("invalid shape type");
-			return importFromWkbPolyline(true, importFlags, true, true,
-					wkbHelper);
-
-		case WkbGeometryType.wkbMultiPoint:
-			if (type.value() != Geometry.GeometryType.MultiPoint
-					&& type.value() != Geometry.GeometryType.Unknown)
-				throw new GeometryException("invalid shape type");
-			return importFromWkbMultiPoint(importFlags, false, false, wkbHelper);
-
-		case WkbGeometryType.wkbMultiPointM:
-			if (type.value() != Geometry.GeometryType.MultiPoint
-					&& type.value() != Geometry.GeometryType.Unknown)
-				throw new GeometryException("invalid shape type");
-			return importFromWkbMultiPoint(importFlags, false, true, wkbHelper);
-
-		case WkbGeometryType.wkbMultiPointZ:
-			if (type.value() != Geometry.GeometryType.MultiPoint
-					&& type.value() != Geometry.GeometryType.Unknown)
-				throw new GeometryException("invalid shape type");
-			return importFromWkbMultiPoint(importFlags, true, false, wkbHelper);
+	public static boolean hasZ(int geometryType) {
+		return (WkbGeometryType.wkbZ & geometryType) == WkbGeometryType.wkbZ
+				|| (WkbGeometryType.wkbZM & geometryType) == WkbGeometryType.wkbZM;
+	}
 
-		case WkbGeometryType.wkbMultiPointZM:
-			if (type.value() != Geometry.GeometryType.MultiPoint
-					&& type.value() != Geometry.GeometryType.Unknown)
-				throw new GeometryException("invalid shape type");
-			return importFromWkbMultiPoint(importFlags, true, true, wkbHelper);
+	public static boolean hasM(int geometryType) {
+		return (WkbGeometryType.wkbM & geometryType) == WkbGeometryType.wkbM
```

Comment:
It should be rewritten to avoid the bitwise ops, because it is not a bitmask
(geometryType > 2000 && geometryType < 2008) || (geometryType > 3000 && geometryType < 3008)

Reply:
So the bitwise op works, but because it's not a bitmask you find it more confusing? I think it's more clear than the alternative

Reply:
it works because the 1s index for wkbGeometryType descriptor is within the first byte's range of values (0-7), but if there was ever a new type it'd not work. And because of that, for clarity's sake you'd like me to change it. 

I can do that.

Reply:
https://github.com/Esri/geometry-api-java/commit/b0715e090d6c2b2bc20dff4636549a5ebc6e3ccc#diff-eccf896d8ca4931baca88ba5034f8037f64ddcedcc74fa05823e997380529034R207